python_modules/libraries/dagster-gcp/dagster_gcp/bigquery/resources.py

```diff
@@ -96,3 +101,39 @@ def bigquery_resource(context):
     bq_resource = BigQueryResource.from_resource_context(context)
     with bq_resource.get_client() as client:
         yield client
+
+
+def get_data_freshness(
+    context: OpExecutionContext,
+    client: bigquery.Client,
+    dataset_id: str,
+    table_ids: Optional[Sequence[str]] = None,
+) -> dict[str, Optional[float]]:
+    """Get the freshness of the tables in a dataset.
+
+    Args:
+        client: BigQuery client.
+        dataset_id: Dataset ID.
+        table_ids: List of table IDs. If not provided, all tables in the dataset will be checked.
+
+    Returns:
+        Dictionary of table IDs to the freshness of the tables.
+    """
+    context.log.info(f"Getting data freshness for dataset {dataset_id}")
+    try:
+        dataset = client.get_dataset(dataset_id)
+    except:
+        context.log.error(f"Dataset {dataset_id} not retrieved")
```

Comment:
I think I would expect exceptions in this case (and the other try/catch case below).

Reply:
Makes sense. I guess depends on if you would rather fail fast and block everything.

Ultimately, I think this is a weirdness with the fact that we're attempting to retrieve the result for multiple tables. I'm kinda thinking we should go back to just one and let users start there.